src/input-nested-multi-select/input-nested-multi-select.tsx

```diff
@@ -87,47 +87,41 @@ export const InputNestedMultiSelect = <V1, V2, V3>({
     const handleSelectItem = (
         item: CombinedFormattedOptionProps<V1, V2, V3>
     ) => {
-        const { keyPath } = item;
-        const selectedItem = getItemAtKeyPath(keyPath);
-        let newKeyPaths = [...selectedKeyPaths];
+        const selectedItem = getItemAtKeyPath(item.keyPath);
+        let newKeyPaths: string[][] = [];
 
         if (selectedItem.subItems) {
-            const keys = getSubItemKeyPaths(selectedItem, keyPath);
-            const isRemoveAll = keys.every((keyPath) =>
-                newKeyPaths.some(
-                    (key) => JSON.stringify(keyPath) === JSON.stringify(key)
-                )
+            const selectableOptionKeyPaths = getSubItemKeyPaths(
+                selectedItem,
+                item.keyPath
             );
 
-            if (isRemoveAll) {
-                newKeyPaths = newKeyPaths.filter((key) =>
-                    keys.every(
-                        (keyPath) =>
-                            JSON.stringify(key) !== JSON.stringify(keyPath)
-                    )
-                );
-            } else {
-                newKeyPaths = [
-                    ...new Map(
-                        [...newKeyPaths, ...keys].map((k) => [k.join("-"), k])
-                    ).values(),
-                ];
+            const selectedCount = selectedKeyPaths.filter((keyPath) =>
+                keyPath.join("-").startsWith(item.keyPath.join("-"))
```

Comment:
can pull this out into a helper function ex.

```
isSubItem(item: string[], category: string[])
```

also it's safer to do
```
JSON.stringify(keyPath) === JSON.stringify(item.keyPath.slice(0, keyPath.length))
```

to avoid case of `["1-2"]` matching `["1", "2"]`